diff --git a/src/gsmenu/gs_actions.c b/src/gsmenu/gs_actions.c
index 5f6e6b0..b549c9f 100644
--- a/src/gsmenu/gs_actions.c
+++ b/src/gsmenu/gs_actions.c
@@ -19,6 +19,7 @@ extern lv_group_t * default_group;
 
 lv_obj_t * restart_pp;
 lv_obj_t * exit_pp;
+lv_obj_t * gs_support;
 lv_obj_t * gs_reboot;
 int restart_value = 1;
 int exit_value = 2;
@@ -29,6 +30,11 @@ void gs_actions_exit_pp(lv_event_t * event)
     sig_handler(*signal);
 }
 
+void gs_actions_support(lv_event_t * event)
+{
+    run_command("/usr/bin/collect_support_data.sh");
+}
+
 void create_gs_actions_menu(lv_obj_t * parent) {
 
     menu_page_data_t* menu_page_data = malloc(sizeof(menu_page_data_t));
@@ -48,6 +54,9 @@ void create_gs_actions_menu(lv_obj_t * parent) {
     exit_pp = create_button(section, "Exit Pixelpilot");
     lv_obj_add_event_cb(lv_obj_get_child_by_type(exit_pp,0,&lv_button_class),gs_actions_exit_pp,LV_EVENT_CLICKED,&exit_value);
 
+    gs_support = create_button(section, "Generate support package");
+    lv_obj_add_event_cb(lv_obj_get_child_by_type(gs_support,0,&lv_button_class),gs_actions_support,LV_EVENT_CLICKED,menu_page_data);
+
     gs_reboot = create_button(section, "Reboot");
     lv_obj_add_event_cb(lv_obj_get_child_by_type(gs_reboot,0,&lv_button_class),generic_button_callback,LV_EVENT_CLICKED,menu_page_data);
 
