diff --git a/drivers/gpu/drm/panel/panel-simple-dsi.c b/drivers/gpu/drm/panel/panel-simple-dsi.c
index e3c8dcf8c..2325b9582 100644
--- a/drivers/gpu/drm/panel/panel-simple-dsi.c
+++ b/drivers/gpu/drm/panel/panel-simple-dsi.c
@@ -307,7 +307,7 @@ static int panel_simple_get_modes(struct drm_panel *panel,struct drm_connector *
 	if (!mode)
 		return 0;
 
-	ret = of_get_drm_display_mode(panel->dev->of_node, mode, p->desc->bus_format,
+	ret = of_get_drm_display_mode(panel->dev->of_node, mode, (u32 *)&p->desc->bus_format,
 				      OF_USE_NATIVE_MODE);
 	if (ret) {
 		dev_dbg(panel->dev, "failed to find dts display timings\n");
@@ -376,7 +376,6 @@ static int panel_simple_regulator_disable(struct drm_panel *panel)
 static int panel_simple_disable(struct drm_panel *panel)
 {
 	struct panel_simple *p = to_panel_simple(panel);
-	int err = 0;
 
 	if (!p->enabled)
 		return 0;
@@ -467,7 +466,6 @@ static int panel_simple_prepare(struct drm_panel *panel)
 static int panel_simple_enable(struct drm_panel *panel)
 {
 	struct panel_simple *p = to_panel_simple(panel);
-	int err = 0;
 
 	if (p->enabled)
 		return 0;
@@ -522,7 +520,6 @@ static int panel_simple_probe(struct device *dev, const struct panel_desc *desc)
 {
 	struct panel_simple *panel;
 	struct panel_desc *of_desc;
-	const char *cmd_type;
 	u32 val;
 	int err;
 
