diff --git a/src/gsmenu/gs_system.c b/src/gsmenu/gs_system.c
index da3031a..621016b 100644
--- a/src/gsmenu/gs_system.c
+++ b/src/gsmenu/gs_system.c
@@ -13,6 +13,7 @@ enum RXMode RXMODE = WFB;
 
 lv_obj_t * rx_mode;
 lv_obj_t * gs_rendering;
+lv_obj_t * connector;
 lv_obj_t * resolution;
 lv_obj_t * rec_enabled;
 lv_obj_t * rec_fps;
@@ -36,6 +37,7 @@ void gs_system_page_load_callback(lv_obj_t * page)
     reload_switch_value(page,gs_rendering);
     reload_dropdown_value(page,rx_mode);
     RXMODE = lv_dropdown_get_selected(lv_obj_get_child_by_type(rx_mode,0,&lv_dropdown_class));
+    reload_dropdown_value(page,connector);
     reload_dropdown_value(page,resolution);
     reload_dropdown_value(page,rec_fps);
     reload_slider_value(page, video_scale);
@@ -138,6 +140,7 @@ void create_gs_system_menu(lv_obj_t * parent) {
     RXMODE = lv_dropdown_get_selected(lv_obj_get_child_by_type(rx_mode,0,&lv_dropdown_class));
 
     gs_rendering = create_switch(cont,LV_SYMBOL_SETTINGS,"GS Rendering","gs_rendering", menu_page_data,false);
+    connector = create_dropdown(cont,LV_SYMBOL_SETTINGS, "Connector","","connector",menu_page_data,false);
     resolution = create_dropdown(cont,LV_SYMBOL_SETTINGS, "Resolution","","resolution",menu_page_data,false);
     video_scale = create_slider(cont, LV_SYMBOL_SETTINGS, "Video scale factor", "video_scale", menu_page_data, false, 2);
 
