--- a/wfb_ng/services.py	2026-07-08 21:05:07.255717179 +0200
+++ b/wfb_ng/services.py	2026-07-08 21:13:20.798759362 +0200
@@ -126,7 +126,7 @@
                 fec_timeout=cfg.fec_timeout,
                 fec_delay=cfg.fec_delay,
                 link_id=link_id,
-                log_interval=settings.common.log_interval,
+                log_interval=getattr(cfg, 'log_interval', settings.common.log_interval),
                 rcv_buf_size=settings.common.tx_rcv_buf_size,
                 snd_buf_size=settings.common.rx_snd_buf_size,
                 injection_retries=cfg.injection_retries,
@@ -171,7 +171,7 @@
                 key=os.path.join(settings.path.conf_dir, cfg.keypair),
                 rcv_buf_size=settings.common.tx_rcv_buf_size,
                 snd_buf_size=settings.common.rx_snd_buf_size,
-                log_interval=settings.common.log_interval,
+                log_interval=getattr(cfg, 'log_interval', settings.common.log_interval),
                 link_id=link_id)).split() + (wlans if not is_cluster else [])
 
     df = RXProtocol(ant_sel_f, cmd, '%s rx' % (service_name,)).start()
@@ -255,7 +255,7 @@
                    key=os.path.join(settings.path.conf_dir, cfg.keypair),
                    rcv_buf_size=settings.common.tx_rcv_buf_size,
                    snd_buf_size=settings.common.rx_snd_buf_size,
-                   log_interval=settings.common.log_interval,
+                   log_interval=getattr(cfg, 'log_interval', settings.common.log_interval),
                    link_id=link_id)).split() + (wlans if not is_cluster else [])
 
     tx_socket_path = '%s-tx-%s' % (service_name, os.urandom(4).hex())
@@ -283,7 +283,7 @@
                    fec_timeout=cfg.fec_timeout,
                    fec_delay=cfg.fec_delay,
                    link_id=link_id,
-                   log_interval=settings.common.log_interval,
+                   log_interval=getattr(cfg, 'log_interval', settings.common.log_interval),
                    rcv_buf_size=settings.common.tx_rcv_buf_size,
                    snd_buf_size=settings.common.rx_snd_buf_size)).split() + wlans
 
@@ -369,7 +369,7 @@
                    key=os.path.join(settings.path.conf_dir, cfg.keypair),
                    rcv_buf_size=settings.common.tx_rcv_buf_size,
                    snd_buf_size=settings.common.rx_snd_buf_size,
-                   log_interval=settings.common.log_interval,
+                   log_interval=getattr(cfg, 'log_interval', settings.common.log_interval),
                    link_id=link_id)).split() + (wlans if not is_cluster else [])
 
     tx_socket_path = '%s-tx-%s' % (service_name, os.urandom(4).hex())
@@ -397,7 +397,7 @@
                    fec_timeout=cfg.fec_timeout,
                    fec_delay=cfg.fec_delay,
                    link_id=link_id,
-                   log_interval=settings.common.log_interval,
+                   log_interval=getattr(cfg, 'log_interval', settings.common.log_interval),
                    rcv_buf_size=settings.common.tx_rcv_buf_size,
                    snd_buf_size=settings.common.rx_snd_buf_size)).split() + wlans
 
@@ -488,7 +488,7 @@
                        key=os.path.join(settings.path.conf_dir, cfg.keypair),
                        rcv_buf_size=settings.common.tx_rcv_buf_size,
                        snd_buf_size=settings.common.rx_snd_buf_size,
-                       log_interval=settings.common.log_interval,
+                       log_interval=getattr(cfg, 'log_interval', settings.common.log_interval),
                        link_id=link_id)).split() + (wlans if not is_cluster else [])
 
         log.msg('%s RX: %s' % (service_name, ' '.join(cmd_rx)))
@@ -520,7 +520,7 @@
                        fec_timeout=cfg.fec_timeout,
                        fec_delay=cfg.fec_delay,
                        link_id=link_id,
-                       log_interval=settings.common.log_interval,
+                       log_interval=getattr(cfg, 'log_interval', settings.common.log_interval),
                        rcv_buf_size=settings.common.tx_rcv_buf_size,
                        snd_buf_size=settings.common.rx_snd_buf_size)).split() + wlans
         log.msg('%s TX: %s' % (service_name, ' '.join(cmd_tx)))
